### ISSUE TITLE
SOTM 2022 Workshop

### ISSUE THREAD

**Issue by hulsiejames** [udsleeds/openinfra#106]

**Comment by GretaTimaite** [comment]
All the tags you will need to run all the openinfra functions. Also see #98 
```{r}
all_extra_tags = c("foot", "bicycle", "access", "service", "maxspeed", "oneway",
                   "kerb", "footway", "sidewalk", "cycleway", "segregated", "highway",
                   "crossing", "lit", "tactile_paving", "surface", "smoothness", "width",
                   "est_width", "lit_by_led", "ref")
```

**Comment by GretaTimaite** [comment]
Participant's issue: 
![image](https://user-images.githubusercontent.com/84614994/185781542-56d47b91-2b5b-48c0-8d8f-78fdbd9e8f6d.png)

I tried on my mac and this is not an issue that I get.  My assumption is that it's because of a weak-ish internet connection (that's usually when I get this issue).

If you still get the same issue, I'd go for this (ie delete repos argument):

```{r}
pkgs = c(
  "tmap",    # package for map making
  "sf",      # geographic vector data classes and functions
  "dplyr",   # data manipulation
  "remotes"  # for installing packages from GitHub
  )

install.packages(pkgs)
```

**Comment by donalhunt** [comment]
I think the issues are related to:
- dependencies missing. might need to add `, dependencies = TRUE` to the `install.packages()` command in R
- missing OS dependencies (I'm seeing a bunch of "please run yum install / apt install" messages.

Will update once I confirm. :+1:

**Comment by GretaTimaite** [comment]
Great. You can also install one pckage at a time, maybe it will help to figure out which one is causing an issue (or maybe all).

Code example

'''{r}
install.packages("sf")
'''

**Comment by donalhunt** [comment]
managed to install all the debian packages + r packages now I think...

but now getting a different error:

```
> tm_shape(data |> dplyr::select("highway")) + 
    tm_lines(col = "highway", title.col = "OSM Highways") + 
    tm_layout( legend.bg.alpha = 0.5, legend.bg.color = "white")
Error: unexpected '>' in "tm_shape(data |>"
```

**Comment by GretaTimaite** [comment]
Yeah, it looks like base R pipe wasn't liked. This should be ok :

```{r}
tm_shape(data["highway"]) + 
  tm_lines(col = "highway", title.col = "OSM Highways") + 
  tm_layout( legend.bg.alpha = 0.5, legend.bg.color = "white")
```

or simply change base R pipe |> to tidy pipe %>% for which load library(tidyverse) loaded:
```{r}
tm_shape(data %>% dplyr::select("highway")) + 
  tm_lines(col = "highway", title.col = "OSM Highways") + 
  tm_layout( legend.bg.alpha = 0.5, legend.bg.color = "white")
```

**Comment by hulsiejames** [comment]
> UPDATE: looks like Debian only ships with R 4.0 and the "|>" (aka "R pipe") is a new construct in R 4.1. currently installing R 4.2 from the relevant repo. grinning

Hoping the upgrade to 4.2 has worked, do let us know if you have any more issues.

**Comment by donalhunt** [comment]
Quick update that after I jumped through all the hoops, I managed to generate the test images.
working through the workshop now. :)

**Comment by hulsiejames** [comment]
> Quick update that after I jumped through all the hoops, I managed to generate the test images.
> 
> working through the workshop now. :)

Great news! Were there any particularly tricky fixes you need to apply? (In case someone has the same issue!) 

Hope it all goes well, and if you have and more issues for feedback more than happy to help and accept constructive criticism!
